abel/linbasex.py

```diff
@@ -0,0 +1,418 @@
+    """wrapper function for linebasex to process supplied quadrant-image as a full-image.
+
+    PyAbel transform functions operate on the right side of an image.
+    Here we follow the `basex` technique of duplicating the right side to
+    the left re-forming the whole image.
+
+    """
+    Dat = np.atleast_2d(Dat)
+
+    quad_rows, quad_cols = Dat.shape
+    full_image = abel.tools.symmetry.put_image_quadrants((Dat, Dat, Dat, Dat),
+                           original_image_shape=(quad_rows*2-1, quad_cols*2-1))
+
+    # inverse Abel transform
+    recon, radial, Beta, QLz = linbasex_transform_full(full_image, an=an, un=un,
+                                 inc=inc, sig_s=sig_s, basis_dir=basis_dir,
+                                 threshold=threshold, clip=clip,
+                                 norm_range=norm_range,
+                                 verbose=verbose, **kwargs)
+
+    # unpack right-side
+    inv_Dat = abel.tools.symmetry.get_image_quadrants(recon)[0]
+
+    if return_Beta:
+        return inv_Dat, radial, Beta, QLz
+    else:
+        return inv_Dat
+
+
+def linbasex_transform_full(Dat, an=[0, 90], un=[0, 2], inc=1, sig_s=0.5,
+                            rcond=0.0005, threshold=0.2, clip=0, basis_dir='./',
+                            return_Beta=False, norm_range=(0, -1),
+                            direction="inverse", verbose=False, **kwargs):
+    """interface function that fetches/calculates the Basis and
+       then evaluates the linbasex inverse Abel transform for the image.
+
+    """
+
+    Dat = np.atleast_2d(Dat)
+
+    rows, cols = Dat.shape
+
+    if rows != cols:
+        raise ValueError('image has shape ({}, {}), '.format(rows, cols) +
+                         'must be square for a "linbasex" transform')
+
+    # generate basis or read from file if available
+    Basis = abel.tools.basis.get_bs_cached("linbasex", cols,
+                  basis_dir=basis_dir,
+                  basis_options=dict(an=an, un=un, inc=inc, clip=clip,
+                                     verbose=verbose))
+
+    return _linbasex_transform_with_basis(Dat, Basis, an=an, un=un, inc=inc,
+                                          rcond=rcond, sig_s=sig_s,
+                                          threshold=threshold, clip=clip,
+                                          norm_range=norm_range)
+
+
+def _linbasex_transform_with_basis(Dat, Basis, an=[0, 90], un=[0, 2], inc=1,
+                                   rcond=0.0005, sig_s=0.5, threshold=0.2,
+                                   clip=0, norm_range=(0, -1)):
+    """linbasex inverse Abel transform evaluated with supplied basis set Basis.
+
+    """
+
+    Dat = np.atleast_2d(Dat)
+
+    rows, cols = Dat.shape
+
+    # Number of used polynoms
+    pol = len(un)
+
+    # How many projections
+    proj = len(an)
+
+    QLz = np.zeros((proj, cols))  # array for projections.
+
+    # Rotate and project VMI-image for each angle (as many as projections)
+    if an == [0, 90]:
```

Comment:
Is the scipy.interpolation.rotate() really that slow? In the interest of keeping the code simpler and easier to debug, it might be nice to eliminate these lines.